Sources/GRPCNIOTransportCore/Client/Connection/Connection.swift

```diff
@@ -129,7 +135,10 @@ package final class Connection: Sendable {
   package func run() async {
     func establishConnectionOrThrow() async throws(RPCError) -> HTTP2Connection {
       do {
-        return try await self.http2Connector.establishConnection(to: self.address)
+        return try await self.http2Connector.establishConnection(
+          to: self.address,
+          authority: self.authority ?? self.address.sniHostname
```

Comment:
Here you use `self.authority ?? self.address.sniHostname`, but further down the file you use `self.authority ?? self.address.authority` - is that intentional?